FreeRTOS/Test/CMock/message_buffer/message_buffer_utest.c

```diff
@@ -328,25 +361,31 @@ void test_xMessageBufferCreate_malloc_fail( void )
  */
 void test_xMessageBufferCreate_zero_size( void )
 {
-    if( TEST_PROTECT() )
-    {
-        ( void ) xMessageBufferCreate( 0 );
-    }
+    xMessageBuffer = xMessageBufferCreate( 0 );
 
-    validate_and_clear_assertions();
+    validate_and_clear_assertion();
+
+    TEST_ASSERT_NOT_NULL( xMessageBuffer );
+
+    TEST_ASSERT_EQUAL( 1U + sizeof( StaticMessageBuffer_t ), dynamicMemoryAllocated );
+
+    vStreamBufferDelete( xMessageBuffer );
 }
 
 /**
  * @brief Should assert if the size passed is less than the minimum size requried to store metadata size.
  */
 void test_xMessageBufferCreate_invalid_size( void )
 {
-    if( TEST_PROTECT() )
-    {
-        ( void ) xMessageBufferCreate( TEST_MESSAGE_METADATA_SIZE );
-    }
+    xMessageBuffer = xMessageBufferCreate( TEST_MESSAGE_METADATA_SIZE );
```

Comment:
Same here. Test should not continue after the assertion is hit.

Reply:
Asserts can be turned off in production. This tests both the configASSERT disabled and configASSERT enabled cases.